### DISCUSSION TITLE
French Paintings Catalogue and NAMA Quire theme

### DISCUSSION THREAD

**Discussion by mgray4** [thegetty/quire#43] [Show & Tell]
The Nelson-Atkins Museum of Art in Kansas City, MO, recently launched its first digital catalogue. It features 106 French paintings and pastels, including the 29 Impressionist and Post-Impressionist masterpieces from the Marion and Henry Bloch Collection given to the museum in 2015. The French Paintings Catalogue can be found on the museum’s website, www.nelson-atkins.org/fpc, where high-resolution images and rich historical context about each painting are freely accessible to all. We launched the catalogue with four paintings, an essay on the history of collecting, and a fun timeline that highlights significant historic events in Kansas City and at the museum. We will add more painting entries each month, sharing new insights and discoveries.

At the core of our digital publication is the Getty’s Quire framework, upon which we built a more robust, searchable platform for an extensive catalogue. Through the incorporation of an adjustable reading space and image viewer, along with enhanced footnotes and glossary terms, we have created an accessible publication. Collection-wide and individual essays and our custom-built timeline guide the reader through the museum’s French paintings collection. We hope by offering our [NAMA theme on GitHub](https://github.com/Nelson-Atkins-Museum/NAMA-Quire-Theme) that we can help Quire become an institutional standard for the digital humanities.

**Comment by geealbers** [comment]
This is so exciting! I know the tremendous amount of work that's gone into this and really love the customizations the Nelson-Atkins team has made here. For anyone not familiar with this project already, be sure to check out one of their enhanced catalogue entries like for [Edouard Manet’s, *The Croquet Party*](https://nelson-atkins.org/fpc/nineteenth-century-realism-barbizon/522/) to see some of the special features they've built.

- Adjustable width text and image frames
- Image comparison tool
- Single-page PDF download
- Accordion-style text navigation

And more!

<img width="1528" alt="Screen Shot 2021-03-11 at 10 24 13 AM" src="https://user-images.githubusercontent.com/7796401/110835953-516e9f00-8254-11eb-8bc3-adffa73b8042.png">

And they're way ahead of the curve on this one, but it's perhaps even more exciting that they're releasing their theme on GitHub for others to reuse! The Quire team will be looking at how we can best help pass that shared work on to the community. Really great!

**Reply by kristhayer11** [reply]
This is awesome! I love that this project is being released in phases, indicated by an icon that signifies a completed object entry. I would like very much to incorporate a couple of the custom features into my current Quire project, particularly a single-page PDF download! Also, I wonder if the code used for accordion-style text could be adapted for a long TOC (instead of using nested TOCs that originate from a folder structure, which I believe you are working on, right @geealbers?)

**Reply by geealbers** [reply]
Oh, good idea @kristhayer11. Using an accordion in TOCs would be really cool! 🤔

**Reply by mmpadilla15** [reply]
@geealbers Have accordions in TOCs been implemented yet? If so, where can I find info on how to do that?

**Reply by geealbers** [reply]
Hi @mmpadilla15, unfortunately not. That feature was never on our roadmap. We were only working on accordions for the main page content, which is documented at https://quire.getty.edu/docs-v1/accordions/.

**Comment by sophielamb** [comment]
This is a fantastic project @mgray4 and you've put so much work in to your customisations.
 
I'm really interested in the Material Icon you've used to mark texts published with scholarly research, and would like to implement something similar to mark peer reviewed articles in our project. 
I've managed to place an icon in the head section of the contents page and sidebar, but am scratching my head about how to attach it to the titles as the contents and sidebar navigation pulls the title data from each essay page. What I want to ask is, where did you put the code for the icon? - did you need to write a partial? Also, how did you place the icon into the contents?

Thank you!

**Reply by geealbers** [reply]
@sophielamb, yes, that's a nice feature. They accomplished it by adding some logic to their `contents-list.html` partial, [lines 207–211](https://github.com/Nelson-Atkins-Museum/NAMA-Quire-Theme/blob/b10426a1151652df57b9829b716b21d5f96b8e34/nelson-atkins/layouts/partials/contents-list.html#L207-L211):

```html
      {{ if or (eq .Type "entry") (eq .Type "entry-dual") }}
        {{ if eq (.Scratch.Get "dateClass") " new"}}
          <i class="material-icons">description</i>
        {{ end }}
      {{ end }}
```

This is particular to their project, but you could achieve the same by pairing a version of this logic with a custom parameter in the page YAML. So, in the YAML of any essay that had been peer reviewed, you'd add:

```yaml
peer_review: true
```

And then in the `contents-list.html` partial, the logic would be:

```html
      {{ if eq .Page.Params.peer_review "true" }}
          <i class="material-icons">description</i>
      {{ end }}
```

Let me know if that works!

**Reply by sophielamb** [reply]
I paired a YAML parameter in the essay pages with the following code in `contents-list.html` in the logic for `{{- define "contents-item-menu" }}`:

    {{ if eq .Page.Params.peer_review "true" }}
        <img src="../img/feather-pen.png" alt="quill icon" width="20" height="20">
    {{ end }}

This sort of works. If I am on the contents page and open the side menu, the icons appear next to the titles. However, if I am on an essay page (or any page that isn't contents) then a broken image link icon is displayed. Why wouldn't the image file be accessed on pages other than the contents?

![image](https://user-images.githubusercontent.com/69150936/121866140-f2ab3c80-ccfe-11eb-8862-a9c227d3440e.png)

Edit: 
The image link is broken when I'm just on an essay page, ie a subdirectory. The easy way around this would be to remove the subdirectories for all markdown files.

**Comment by mmpadilla15** [comment]
Hi all - Newbee here, so apologies for not entirely understanding how everything works!

I've been scouring Quire pubs looking for examples of timelines and the Nelson-Atkins epub has a nice one. I went into the Layouts folder and found a file called Timeline, but it's an HTML file. Is there a MD file somewhere that I can borrow/steal to try our in the epub I'm working on? Or is that even what I need to incorporate at timeline into my project?

**Reply by mgray4** [reply]
Hi Michele, Glad you're interested in our timeline! @salger1 would be the best person to answer that; perhaps you can chat next week?

**Reply by mmpadilla15** [reply]
Thanks for the response @mgray4! I would love to chat with you and/or @salger1 sometime. If you'd like to email me, I'm at michellep@cartermuseum.org.

**Comment by ncamachoh** [comment]
Hi!

I am working on a new project and was wondering if anyone can help me make accordion menus work!
I have managed to add the html tags and css but for some reason they won't actually expand, so I just end up with titles in boxes, without any of the content.

I tried adding the js to the _assets> javascript >/*custom.js page, but it won't read it, and if I add it directly under the content it breaks the preview!

I am not a developer so this is were I got the instructions: https://www.w3schools.com/howto/howto_js_accordion.asp 

Thanks!

**Reply by Erin-Cecele** [reply]
Hi @ncamachoh. Thank you for asking this question. Would you mind posting it as a separate thread? This is a valuable topic that others would likely benefit from knowing the answer to, and it will be more easily searchable as its own post. Thank you!
